src/validator.js

```diff
@@ -271,14 +360,22 @@ export default class Validator {
         })
         .then(() => {
           this.xpi = new _Xpi(this.packagePath);
+          return this.detectPackageType();
+        })
+        .then((packageType) => {
+          this.packageType = packageType;
+          if (packageType) {
+            log.debug('Package type detected as %s',
```

Comment:
While in this particular case the formatting probably _does_ look nicer than having to include `singleLineString`, we should be consistent and use template strings everywhere rather than `%s`.